### PR TITLE
Fix bug #17445

### LINKED ISSUE #17445: Issue with exporting using Firefox or Safari (or PHP8.1.4 issue) - file is not saved because of issue with reading source file
### Describe the bug

Not sure if it's FF or phpMyAdmin or something completely different but recently whenever I want to export anything (doesn't matter if it's sql or csv and if it's compressed or not) I'm getting weird message about FF not being able to read source file, which results in file not being downloaded/saved. The file placeholder is created at the destination but it's removed right after hitting confirmation on the window:
obraz
The message in the window can be translated as:
```
Downloading ...
File ... can not be saved because source file is not readable.
You should try later or contact server administrator.
```
I also tried that on older FF version (98dev) and there was no message but the file was empty.
Using `View output as file` works properly.

Also affects Safari: https://github.com/hestiacp/hestiacp/issues/2529#issuecomment-1099876177

### To Reproduce

1. Go to any table.
2. Click on Export.
3. Choose to save as file.
4. See error.

### Server configuration

- Operating system: ArchLinux
- Web server: Apache/2.4.52
- Database version: doesn't matter, tried this on few
- PHP version: 8.1.4
- phpMyAdmin version: 5.1.3

### Client configuration

- Browser: Firefox 99.0b5
- Operating system: Win10

### PR DESCRIPTION
**Pull request by kamil-tekiela**
Fixes #17445 

Could someone help me with writing a test case for this?

### CODE REVIEW

File: libraries/classes/Core.php [hunk 1]
```diff
@@ -635,10 +636,15 @@ public static function downloadHeader(
...
-            header('Content-Encoding: gzip');
+        if (strpos($mimetype, 'gzip') !== false) {
+            $notChromeOrLessThan43 = PMA_USR_BROWSER_AGENT != 'CHROME' || PMA_USR_BROWSER_VER < 43;// see bug #4942
+            if ($notChromeOrLessThan43) {
+                header('Content-Encoding: gzip');
+            }
+        } else {
+            // The default output in PMA uses gzip,
+            // so if we want to output uncompressed file, we should reset the encoding
+            header_remove('Content-Encoding');
```
Comment (williamdes) [inline libraries/classes/Core.php:647]: Can you add the php bug link?

Comment (kamil-tekiela) [reply libraries/classes/Core.php:647]: Let's wait to see what will result from the PHP bug report. I don't know which bug to refer to here.

Comment (henri9813) [reply libraries/classes/Core.php:647]: Hello,

Do you know when this gonna be release ?

We have many customers impacted by this bugs and could be helped thanks this patch

Best regards

Comment (williamdes) [reply libraries/classes/Core.php:647]: Bonjour Henri,

I can not give a date for the release but this fix will be shipped in the next 5.1 release.
When it is merged you can try it with snapshots: [(phpMyAdmin 5.1+snapshot)](https://www.phpmyadmin.net/downloads/#snapshot_5.1+snapshot)
Will that be okay for you in the mean time ?

**Comment by williamdes** [comment]
> Could someone help me with writing a test case for this?

Sure, I will write a test after merging this. Thanks for the patience

**Comment by mauriciofauth** [comment]
> Could someone help me with writing a test case for this?

@kamil-tekiela, I added tests for `PhpMyAdmin\Core::downloadHeader` in 924edd57fa243d909a6ec671b3625311c6468c65. You can rebase with `QA_5_1` to have it.

**Review by williamdes — APPROVED** [review]

**Comment by kamil-tekiela** [comment]
> > Could someone help me with writing a test case for this?
> 
> @kamil-tekiela, I added tests for `PhpMyAdmin\Core::downloadHeader` in [924edd5](https://github.com/phpmyadmin/phpmyadmin/commit/924edd57fa243d909a6ec671b3625311c6468c65). You can rebase with `QA_5_1` to have it.

Thank you. It seems like something is missing in the new test.

**Comment by mauriciofauth** [comment]
> Thank you. It seems like something is missing in the new test.

I didn't run the full testsuite with PHP 8. Fixed now.

**Comment by osxtra** [comment]
This could be something else in my environment, but the gzip issue remains when attempting to export to OpenDocument binaries.

PHPmyAdmin: 5.1.3
PHP: 8.1.4 (compiled with libzip 1.8.0)
OS:  OSX 10.15.7

Tested with current versions of FireFox, Opera, Waterfox, and SeaMonkey.
Not tested with Chrome or Edge.

Gleaned from willimades's post of 2022-03-18, commenting out lines 641 ( "header('Content-Encoding: gzip');"  ) and 643 ( "header('Content-Transfer-Encoding: binary');" ) allowed binary exports to work again.

**Comment by williamdes** [comment]
> This could be something else in my environment, but the gzip issue remains when attempting to export to OpenDocument binaries.
> 
> PHPmyAdmin: 5.1.3 PHP: 8.1.4 (compiled with libzip 1.8.0) OS: OSX 10.15.7
> 
> Tested with current versions of FireFox, Opera, Waterfox, and SeaMonkey. Not tested with Chrome or Edge.
> 
> Gleaned from willimades's post of 2022-03-18, commenting out lines 641 ( "header('Content-Encoding: gzip');" ) and 643 ( "header('Content-Transfer-Encoding: binary');" ) allowed binary exports to work again.

Hi @osxtra 
Can you try the latest 5.2 version in development [(phpMyAdmin 5.2+snapshot)](https://www.phpmyadmin.net/downloads/#snapshot_5.2+snapshot) and let me know if this is fixed ?
Else you can discuss about this on #16042

**Pull request merged**